modules/simple_forms_api/spec/services/s3/submission_archive_builder_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require SimpleFormsApi::Engine.root.join('spec', 'spec_helper.rb')
+
+RSpec.describe SimpleFormsApi::S3::SubmissionArchiveBuilder do
+  let(:form_id) { '21-10210' }
+  let(:form_data) { File.read("modules/simple_forms_api/spec/fixtures/form_json/vba_#{form_id.gsub('-', '_')}.json") }
+  let(:submission) { create(:form_submission, :pending, form_type: form_id, form_data:) }
+  let(:benefits_intake_uuid) { submission.benefits_intake_uuid }
+  let(:archive_builder_instance) { described_class.new(benefits_intake_uuid:) }
+
+  before do
+    allow(FormSubmission).to receive(:find_by).and_return(submission)
+    allow_any_instance_of(described_class).to receive(:assign_instance_variables).and_call_original
+    allow(SecureRandom).to receive(:hex).and_return('random-letters-n-numbers')
+  end
+
+  describe '#initialize' do
+    subject(:new) { archive_builder_instance }
+
+    let(:defaults) do
+      {
+        attachments: [],
+        file_path: nil,
+        include_json_archive: true,
+        include_manifest: true,
+        include_text_archive: true,
+        metadata: {},
+        parent_dir: 'vff-simple-forms'
+      }
+    end
+
+    it { is_expected.to have_received(:assign_instance_variables).with(defaults) } # rubocop:disable RSpec/SubjectStub
+  end
+
+  describe '#run' do
```

Comment:
I know this is just cursory at the moment but it would be great later if we could have more tests, perhaps these?
- Toggle `include_json_archive` and ensure `File.write` does/does not get called with the file name.
- Toggle `include_text_archive` and ensure `File.write` does/does not get called with the file name.
- Toggle `write_manifest` and ensure `CSV.open` does/does not get called.
- Similar for attachments.
^ these would be a good start, but we can work on them as a followup.

Reply:
Oh yeah, i want WAY more test coverage. These are great suggestions! I just wanted to get this out so we could start testing it on staging. I'll be adding more coverage for sure